zdx/writer.go

```diff
@@ -28,102 +31,192 @@ import (
 //
 // As the zng file data is written, a b-tree index is computed as a
 // constant b-tree to make key lookups efficient.  The b-tree sections
-// are written to temporary zng files (.1, .2, etc) and at close,
-// they are collapsed into the single-file zdx format.
-// XXX TBD: the single-file implementation will arrive in a subsequent PR.
+// are written to temporary files and at close, they are merged into
+// a single-file microindex.
 type Writer struct {
 	zctx        *resolver.Context
-	uri         iosrc.URI
-	level       int
-	writer      *bufwriter.Writer
-	out         *zngio.Writer
-	parent      *Writer
-	header      *zng.Record
-	frameThresh int
-	frameStart  int64
-	frameEnd    int64
-	frameKey    *zng.Record
-	keyFields   []string
+	writer      *indexWriter
 	cutter      *proc.Cutter
-	recType     *zng.TypeRecord
+	tmpdir      string
+	frameThresh int
+	keyType     *zng.TypeRecord
+	iow         io.WriteCloser
+	childField  string
 }
 
-// NewWriter returns a Writer ready to write an zdx bundle and related
-// index files via subsequent calls to Write(), or it returns an error.
-// All files will be written to the directory indicated by path
-// with the form $path, $path.1, and so forth.  Calls to Write must
+type indexWriter struct {
+	base       *Writer
+	parent     *indexWriter
+	name       string
+	buffer     *bufwriter.Writer
+	zng        *zngio.Writer
+	frameStart int64
+	frameEnd   int64
+	frameKey   *zng.Record
+	headerSize int
+}
+
+// NewWriter returns a Writer ready to write a microindex or it returns
+// an error.  The microindex is written to the URL given while temporary
+// file are written locally.  Calls to Write must
 // provide keys in increasing lexicographic order.  Duplicate keys are not
 // allowed but will not be detected.  Close() must be called when done writing.
-func NewWriter(zctx *resolver.Context, path string, keyFields []string, framesize int) (*Writer, error) {
+func NewWriter(zctx *resolver.Context, path string, keyFields []string, frameThresh int) (*Writer, error) {
```

Comment:
If frameThresh is a better name (I'm still catching up with all of this...), should we change the various places that still refer to `framesize`? I'm primarily wondering about CLI docs...

Reply:
Yes, I'll update.